sympy/core/tests/test_subs.py

```diff
@@ -640,3 +640,13 @@ def test_issue_5217():
     assert z.subs(sub) == 1 - s
     assert q == 4*x**2*y**2
     assert q.subs(sub) == 2*y**2*s
+
+
+def test_pow_eval_subs_no_cache():
+    from sympy.core.cache import clear_cache
+    from sympy.abc import x, y
+
+    s = 1/sqrt(x**2)
+    clear_cache()
```

Comment:
I think tests do `clear_cache` anyway. Doctests definitely do, and the code that called `clear_cache` looked like it was supposed to be called for both doctests and normal tests, but I haven't verified.
I think the issue we're having is with the caching that happens within a single test.